nvflare/app_common/workflows/hierarchical_statistics_controller.py

```diff
@@ -0,0 +1,288 @@
+
+        .. code-block:: text
+
+            "statistic_configs": {
+                "count": {},
+                "mean": {},
+                "stddev": {}
+            }
+
+        Only count, mean and stddev statistics are specified, so the statistics_controller
+        will only set tasks to calculate these three statistics.
+
+        For 'hierarchy_config', below is an example hierarchy specification with 4 level hierarchy for 9 NVFLARE clients with the names ranging
+        from 'Device-1' to 'Device-9' and with hierarchical levels named 'Manufacturers', 'Orgs', 'Locations', 'Devices' with
+        'Manufacturers' being the top most hierarchical level and "Devices" being the lowest hierarchical level:
+
+        .. code-block:: text
+           {
+                "Manufacturers": [
+                    {
+                    "Name": "Manufacturer-1",
+                    "Orgs": [
+                        {
+                        "Name": "Org-1",
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-1", "Device-2"]
+                            },
+                            {
+                            "Name": "Location-2",
+                            "Devices": ["Device-3"]
+                            }
+                        ]
+                        },
+                        {
+                        "Name": "Org-2",
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-4", "Device-5"]
+                            },
+                            {
+                            "Name": "Location-2",
+                            "Devices": ["Device-6"]
+                            }
+                        ]
+                        }
+                    ]
+                    },
+                    {
+                    "Name": "Manufacturer-2",
+                    "Orgs": [
+                        {
+                        "Name": "Org-3",
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-7", "Device-8"]
+                            },
+                            {
+                            "Name": "Location-6",
+                            "Devices": ["Device-9"]
+                            }
+                        ]
+                        }
+                    ]
+                    }
+                ]
+            }
+
+        """
+        super().__init__(
+            statistic_configs,
+            writer_id,
+            wait_time_after_min_received,
+            result_wait_timeout,
+            precision,
+            min_clients,
+            enable_pre_run_task,
+        )
+        self.hierarchy_config = hierarchy_config
```

Comment:
Please validate hierarchy_config to make sure it's a valid file name. Raise exception if not.

Reply:
Done, thanks.